src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -108,6 +132,39 @@ public void deleteTag(Tag tag) throws PersonNotFoundException, DuplicatePersonEx
         addressBook.removeTag(tag);
     }
 
+    //@@author jelneo
+    /**
+     * Authenticates user
+     * @throws UserNotFoundException if username and password does not match those in the user preference file
+     * @throws IllegalValueException if username and password does not meet username and password requirements
+     */
+    public void authenticateUser(Username username, Password password) throws UserNotFoundException,
+            IllegalValueException {
+        Username fileUsername = new Username(getUsernameFromUserPref());
+        Password filePassword = new Password(getPasswordFromUserPref());
+        if (fileUsername.equals(username) && filePassword.equals(password)) {
+            raise(new LoginAppRequestEvent(true));
+        } else {
+            raise(new LoginAppRequestEvent(false));
+            throw new UserNotFoundException();
+        }
+    }
+
+    public String getUsernameFromUserPref() {
+        return userPrefs.getAdminUsername();
+    }
+
+    public String getPasswordFromUserPref() {
+        return userPrefs.getAdminPassword();
+    }
+
+    //@@author
```

Comment:
Try not to have dangling, random comments in the code base ^^.

Reply:
It is to signify that the code from starting from `//@@author jelneo` up until that tag was written by me. The code below that tag was originally in the addressbook-level4` repo.